### PR TITLE
App refresh indicator

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)
- Changes the behaviour of the refresh indicator to make the specifications on the Quality assurance checklist

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [x] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
* Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?
- [MOB-66](https://airqoteam.atlassian.net/browse/MOB-66)

#### Screenshots (optional)
Screenshot_1648543737 (1)
Simulator Screen Shot - iPhone 8 - 2022-03-29 at 11 53 31 (1)

### CODE REVIEW

File: mobile/lib/widgets/custom_shimmer.dart [hunk 1]
```diff
@@ -47,6 +48,24 @@ void loadingScreen(BuildContext _context) async {
...
+        refreshTriggerPullDistance: 70,
+        refreshIndicatorExtent: 60,
+        onRefresh: onRefresh,
+      ),
+      SliverList(
+        delegate: sliverChildDelegate,
+      ),
+    ],
+  );
+}
```
Comment (NoahNsimbe) [inline mobile/lib/widgets/custom_shimmer.dart:67]: Refresh indicator widget

File: mobile/lib/screens/settings_page.dart [hunk 2]
```diff
@@ -254,7 +277,7 @@ class _SettingsPageState extends State<SettingsPage> {
             onTap: () async {
               await _rateService.rateApp();
             },
-            child: cardSection('Rate the AirQo App'),
+            child: _cardSection('Rate the AirQo App'),
```
Comment (NoahNsimbe) [inline mobile/lib/screens/settings_page.dart:280]: replacing public methods with private methods

File: mobile/lib/screens/notification_page.dart [hunk 3]
```diff
@@ -75,19 +62,17 @@ class _NotificationPageState extends State<NotificationPage> {
...
-          child: ListView.builder(
-            itemBuilder: (context, index) => Padding(
-              padding: const EdgeInsets.fromLTRB(10, 5, 10, 5),
-              child: notificationCard(_notifications[index]),
-            ),
-            itemCount: _notifications.length,
-          ),
-        ));
+        child: refreshIndicator(
+            sliverChildDelegate: SliverChildBuilderDelegate((context, index) {
```
Comment (NoahNsimbe) [inline mobile/lib/screens/notification_page.dart:66]: using a custom refresh indicator

File: mobile/lib/screens/analytics_view.dart [hunk 4]
```diff
@@ -14,81 +15,44 @@ class AnalyticsView extends StatefulWidget {
...
-                      height: 15,
-                    ),
-                    containerLoadingAnimation(height: 120.0, radius: 16.0),
-                    const SizedBox(
-                      height: 15,
-                    ),
-                    containerLoadingAnimation(height: 120.0, radius: 16.0),
-                  ],
-                );
-              }
```
Comment (NoahNsimbe) [inline mobile/lib/screens/analytics_view.dart:80]: reducing code repetition

**Review by Baalmart — APPROVED** [review]
Thanks @NoahNsimbe , the changes LGTM.

In the future, please include the steps to be done in order to setup and test these mobile app PRs. Unless it is not practical for this case?

**Pull request merged**